src/client/csfle/state_machine.rs

```diff
@@ -181,6 +216,48 @@ impl CryptExecutor {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct MongocryptdOptions {
+    pub(crate) spawn_path: Option<PathBuf>,
+    pub(crate) spawn_args: Vec<String>,
+    pub(crate) uri: Option<String>,
+}
+
+impl MongocryptdOptions {
+    async fn spawn(&self) -> Result<Client> {
+        use std::process::{Command, Stdio};
+
+        let which_path;
+        let bin_path = match &self.spawn_path {
+            Some(s) => s,
+            None => {
+                which_path = which::which("mongocryptd")
+                    .map_err(|e| Error::invalid_argument(format!("{}", e)))?;
+                &which_path
+            }
+        };
+        let mut args: Vec<&str> = vec![];
+        let mut has_idle = false;
+        for arg in &self.spawn_args {
+            has_idle |= arg.starts_with("--idleShutdownTimeoutSecs");
+            args.push(arg);
+        }
+        if !has_idle {
+            args.push("--idleShutdownTimeoutSecs=60");
+        }
+        Command::new(bin_path)
```

Comment:
Spawning a process is potentially blocking, so I think we'll want to use the async runtimes versions of this. Unfortunately, I don't think they're cross-platform, so we'll need to add a new wrapper layer to our `runtime` module. Fortunately, I think the API is the same between the two runtimes:
- [tokio](https://docs.rs/tokio/latest/tokio/process/index.html)
- [async-std](https://docs.rs/async-std/latest/async_std/process/index.html)

Alternatively, we could just use `std`'s API and throw it in a `spawn_blocking` task, but I think using the runtime native APIs would probably be preferable.